File: app/src/main/java/com/minibrowser/security/InputValidator.java
package com.minibrowser.security;

import java.net.MalformedURLException;
import java.net.URL;

public class InputValidator {
    public static boolean isEndpointValid(String url) {
        if (url == null || url.trim().isEmpty()) return false;
        try {
            new URL(url);
            return url.startsWith("http://") || url.startsWith("https://");
        } catch (MalformedURLException e) {
            return false;
        }
    }

    public static boolean isCssValid(String css) {
        if (css == null) return false;
        return !css.contains("javascript:") && !css.contains("<script");
    }

    public static boolean isJsValid(String js) {
        if (js == null) return false;
        return !js.contains("AndroidBlock.") && !js.contains("__mbNav.");
    }

    public static boolean isBlocklistUrlValid(String url) {
        return isEndpointValid(url);
    }
}

File: app/src/main/java/com/minibrowser/security/InputValidator.kt
package com.minibrowser.security

import java.net.MalformedURLException
import java.net.URL

object InputValidator {
    @JvmStatic
    fun isEndpointValid(url: String?): Boolean {
        if (url.isNullOrBlank()) return false
        return try {
            URL(url)
            url.startsWith("http://") || url.startsWith("https://")
        } catch (e: MalformedURLException) {
            false
        }
    }

    @JvmStatic
    fun isCssValid(css: String?): Boolean {
        if (css == null) return false
        return !css.contains("javascript:") && !css.contains("<script")
    }

    @JvmStatic
    fun isJsValid(js: String?): Boolean {
        if (js == null) return false
        return !js.contains("AndroidBlock.") && !js.contains("__mbNav.")
    }

    @JvmStatic
    fun isBlocklistUrlValid(url: String?): Boolean {
        return isEndpointValid(url)
    }
}
